portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,33 +46,46 @@ private String convertToJsonUsingGson(List<String> messages) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get comments in datastore, by most recent order at the top.
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    Query query = new Query("Comment").addSort("date", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    // Iterate over all entities, get comment text.
-    List<String> comments = new ArrayList<>();
+    // Extract limit on number of comments from query string.
+    Integer limit = Integer.parseInt(request.getParameter("limit"));
+
+    // Iterate over all entities, get comment.
+    List<Comment> comments = new ArrayList<>();
+    Integer count = 0;
```

Comment:
You should be able to use the primitive 'int' here instead of the object 'Integer', whenever possible it's best to use the primitive since it uses less memory

Reply:
Got it, that makes sense.